homeassistant/components/comfoconnect/climate.py

```diff
@@ -0,0 +1,136 @@
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_COMFOCONNECT_UPDATE_RECEIVED, ComfoConnectBridge
+
+_LOGGER = logging.getLogger(__name__)
+HVAC_MODES = [HVAC_MODE_HEAT, HVAC_MODE_COOL]
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the ComfoConnect climate platform."""
+    if discovery_info is None:
+        return
+
+    ccb = hass.data[DOMAIN]
+    add_entities([ComfoConnectBypass(ccb.name, ccb)], True)
+
+
+class ComfoConnectBypass(ClimateEntity):
+    """Representation of the ComfoConnect bypass platform."""
+
+    def __init__(self, name, ccb: ComfoConnectBridge) -> None:
+        """Initialize the ComfoConnect bypass."""
+        self._name = name
+        self._ccb = ccb
+        self.sensor_temperature_supply_local = 0
+        self.sensor_bypass_state_local = 0
+
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+        _LOGGER.debug("Registering for bypass state")
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_COMFOCONNECT_UPDATE_RECEIVED.format(SENSOR_BYPASS_STATE),
+                self._handle_update_bypass,
+            )
+        )
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_COMFOCONNECT_UPDATE_RECEIVED.format(SENSOR_TEMPERATURE_SUPPLY),
+                self._handle_update_temp,
+            )
+        )
+        await self.hass.async_add_executor_job(
+            self._ccb.comfoconnect.register_sensor, SENSOR_BYPASS_STATE
+        )
+        await self.hass.async_add_executor_job(
+            self._ccb.comfoconnect.register_sensor, SENSOR_TEMPERATURE_SUPPLY
+        )
+
+    @callback
+    def _handle_update_bypass(self, value):
+        """Handle update callback for bypass state."""
+        _LOGGER.debug(
+            "Handle update for climate - bypass (%d): %s", SENSOR_BYPASS_STATE, value
+        )
+        self.sensor_bypass_state_local = value
+
+        self._ccb.data[SENSOR_BYPASS_STATE] = self.sensor_bypass_state_local
+        self.async_write_ha_state()
+
+    @callback
+    def _handle_update_temp(self, value):
+        """Handle update callback for temperature."""
+        _LOGGER.debug(
+            "Handle update for climate - temp (%d): %s",
+            SENSOR_TEMPERATURE_SUPPLY,
+            value,
+        )
+        self.sensor_temperature_supply_local = value
+
+        self._ccb.data[SENSOR_TEMPERATURE_SUPPLY] = self.sensor_temperature_supply_local
+        self.async_write_ha_state()
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        _LOGGER.debug("set_hvac_mode: %s", hvac_mode)
+
+        if hvac_mode == HVAC_MODE_HEAT:
+            _LOGGER.debug("Send command: CMD_BYPASS_OFF")
+            self._ccb.comfoconnect.cmd_rmi_request(CMD_BYPASS_OFF)
```

Comment:
How come we use `async_add_executor_job` in the constructor when calling the API, but here we call it directly?